hack/lib.sh

```diff
@@ -128,43 +128,44 @@ function enable_workload_identity(){
   gcloud services enable iamcredentials.googleapis.com
   # Enable workload identity.
   echo "Enable Workload Identity"
-  gcloud container clusters update ${cluster_name} \
-    --${cluster_location_option}=${cluster_location} \
-    --workload-pool=${project_id}.svc.id.goog
+  gcloud container clusters update "${cluster_name}" \
+    --${cluster_location_option}="${cluster_location}" \
+    --workload-pool="${project_id}".svc.id.goog
 
   # Modify all node pools to enable GKE_METADATA.
    echo "Modify all node pools to enable GKE_METADATA"
-  pools=$(gcloud container node-pools list --cluster=${cluster_name} --${cluster_location_option}=${cluster_location} --format="value(name)")
+  pools=$(gcloud container node-pools list --cluster="${cluster_name}" --${cluster_location_option}="${cluster_location}" --format="value(name)")
   while read -r pool_name
   do
   gcloud container node-pools update "${pool_name}" \
-    --cluster=${cluster_name} \
-    --${cluster_location_option}=${cluster_location} \
+    --cluster="${cluster_name}" \
+    --${cluster_location_option}="${cluster_location}" \
     --workload-metadata=GKE_METADATA
   done <<<"${pools}"
 
-  gcloud projects add-iam-policy-binding ${project_id} \
-    --member=serviceAccount:${control_plane_service_account}@${project_id}.iam.gserviceaccount.com \
+  gcloud projects add-iam-policy-binding "${project_id}" \
+    --member=serviceAccount:"${control_plane_service_account}"@"${project_id}".iam.gserviceaccount.com \
     --role roles/iam.serviceAccountAdmin
   }
 
 function storage_admin_set_up() {
   echo "Update ServiceAccount for Storage Admin"
   local project_id=${1}
   local pubsub_service_account=${2}
-  local pubsub_service_account_key_temp=${3}
+  local project_number="$(gcloud projects describe "${project_id}" --format="value(projectNumber)")"
+  local access_token="$(gcloud auth application-default print-access-token)"
```

Comment:
Which credential does this command use? The credential for the control plane or the credential for the data plane?
https://cloud.google.com/sdk/gcloud/reference/auth/application-default/print-access-token. 
In the docs, it looks like it may not always be the credential for the data plane.

Reply:
You are right, I have added the service account key back and used it to get the `access_token`.